testSrc/unit/io/flutter/android/AndroidxTest.java

```diff
@@ -0,0 +1,162 @@
+    assertTrue(FlutterUtils.isAndroidxProject(fixture.getProject()));
+  }
+
+  @Test
+  public void androidxModuleRecognized() {
+    createTestHarness(".android", FlutterProjectType.MODULE, ANDROIDX_PROPERTIES);
+    assertTrue(FlutterUtils.isAndroidxProject(fixture.getProject()));
+  }
+
+  @Test
+  public void androidPluginRecognized() {
+    createTestHarness("example/android", FlutterProjectType.PLUGIN, ANDROID_PROPERTIES);
+    assertFalse(FlutterUtils.isAndroidxProject(fixture.getProject()));
+  }
+
+  @Test
+  public void androidxPluginRecognized() {
+    createTestHarness("example/android", FlutterProjectType.PLUGIN, ANDROIDX_PROPERTIES);
+    assertTrue(FlutterUtils.isAndroidxProject(fixture.getProject()));
+  }
+
+  @Test
+  public void packageRecognized() {
+    boolean result = FlutterUtils.isAndroidxProject(fixture.getProject());
+    assertFalse(result);
+  }
+
+  private void createTestHarness(@NotNull String androidPath, @Nullable FlutterProjectType type, @NotNull String propertiesContent) {
+    @SystemIndependent String basePath = fixture.getProject().getBasePath();
+    VirtualFile projectDir = LocalFileSystem.getInstance().findFileByPath(basePath);
+    try {
+      Testing.runOnDispatchThread(() -> {
+        ApplicationManager.getApplication().runWriteAction(() -> {
+          try {
+            if (type != null) {
+              VirtualFile metadataFile = projectDir.createChildData(this, ".metadata");
+              String metadataContent = "project_type: ";
+              switch (type) {
+                case APP:
+                  metadataContent += "app\n";
+                  break;
+                case MODULE:
+                  metadataContent += "module\n";
+                  break;
+                case PACKAGE:
+                  metadataContent += "package\n";
+                  break;
+                case PLUGIN:
+                  metadataContent += "plugin\n";
+                  break;
+                case IMPORT:
+                  fail();
+                  break;
+              }
+              try (OutputStream out = new BufferedOutputStream(metadataFile.getOutputStream(this))) {
+                out.write(metadataContent.getBytes(StandardCharsets.UTF_8));
+              }
+            }
+            String[] dirs = androidPath.split("/");
+            VirtualFile androidDir = projectDir;
+            for (String dir : dirs) {
+              androidDir = androidDir.createChildDirectory(this, dir);
+            }
+            assertNotEquals(androidDir, projectDir);
+            VirtualFile propertiesFile = androidDir.createChildData(this, "gradle.properties");
+            try (OutputStream out = new BufferedOutputStream(propertiesFile.getOutputStream(this))) {
+              out.write(propertiesContent.getBytes(StandardCharsets.UTF_8));
+            }
+          }
+          catch (IOException x) {
+            System.out.println(x.getMessage());
```

Comment:
why aren't we failing the test if there is an IOException?

Reply:
I forgot to replace the debugging code. Done.